fastchat/serve/inference.py

```diff
@@ -160,31 +160,46 @@ def generate_stream(
                 skip_special_tokens=True,
                 spaces_between_special_tokens=False,
             )
+
+            def partial_stop(output, stop_str):
+                for i in range(0, min(len(output), len(stop_str))):
+                    if stop_str.startswith(output[-i:]):
+                        return True
+                return False
+
             if stop_str:
                 if isinstance(stop_str, str):
                     pos = output.rfind(stop_str, rfind_start)
                     if pos != -1:
                         output = output[:pos]
                         stopped = True
+                    else:
+                        partial_stop = partial_stop(output, stop_str)
```

Comment:
here is a bug: this variable assignment overwrites `partial_stop` with a boolean. in the next iteration of the loop this will trigger an error

Reply:
I made `partial_stop()` global, and renamed flag to `partially_stopped`